invenio_accounts/models.py

```diff
@@ -60,6 +61,14 @@
 """Relationship between users and roles."""
 
 
+class CaseInsensitiveComparator(Comparator):
+    """Class allowing case-insensitive comparisons on an attribute."""
+
+    def __eq__(self, other):
+        """Case-insensitive equal operation."""
+        return func.lower(self.__clause_element__()) == func.lower(other)
```

Comment:
thoughts on https://docs.python.org/3/library/stdtypes.html#str.casefold instead?

Reply:
I didn't about `casefold`. :+1:

However, in this case, this `func.lower` is coming from SQLAlchemy and translated to the SQL `LOWER` function.
Since we do `self._username = username.lower()` in the username setter in this same `models.py`, this should be consistent.